internal/daemon/controller/gateway_test.go

```diff
@@ -95,6 +96,111 @@ func Test_correlationIdAnnotator(t *testing.T) {
 	assert.Equal(t, corId, corIds[0])
 }
 
+func Test_clientAgentHeadersAnnotator(t *testing.T) {
+	t.Parallel()
+	t.Run("Valid single user-agent", func(t *testing.T) {
+		t.Parallel()
+		req := &http.Request{
+			Header: map[string][]string{
+				"User-Agent": {"Boundary-client-agent/0.1.4"},
+			},
+		}
+
+		md := userAgentHeadersAnnotator(context.Background(), req)
+		require.NotNil(t, md)
+		values := md.Get("userAgents")
+		require.Len(t, values, 1)
+
+		var parsed []map[string]interface{}
+		err := handlers.JSONMarshaler().Unmarshal([]byte(values[0]), &parsed)
+		require.NoError(t, err)
+		require.Len(t, parsed, 1)
+		assert.Equal(t, "Boundary-client-agent", parsed[0]["product"])
+		assert.Equal(t, "0.1.4", parsed[0]["product_version"])
+	})
+
+	t.Run("Multiple valid agents with comments", func(t *testing.T) {
+		t.Parallel()
+		req := &http.Request{
+			Header: map[string][]string{
+				"User-Agent": {"Boundary-client-agent/0.1.4 (foo; bar); AnotherApp/2.0.0 (baz)"},
+			},
+		}
+
+		md := userAgentHeadersAnnotator(context.Background(), req)
+		require.NotNil(t, md)
+		values := md.Get("userAgents")
+		require.Len(t, values, 1)
+
+		var parsed []map[string]interface{}
+		err := handlers.JSONMarshaler().Unmarshal([]byte(values[0]), &parsed)
+		require.NoError(t, err)
+		require.Len(t, parsed, 2)
+
+		assert.Equal(t, "Boundary-client-agent", parsed[0]["product"])
+		assert.Equal(t, "0.1.4", parsed[0]["product_version"])
+		assert.ElementsMatch(t, []string{"foo", "bar"}, parsed[0]["comments"].([]interface{}))
+
+		assert.Equal(t, "AnotherApp", parsed[1]["product"])
+		assert.Equal(t, "2.0.0", parsed[1]["product_version"])
+		assert.ElementsMatch(t, []string{"baz"}, parsed[1]["comments"].([]interface{}))
+	})
+
+	t.Run("No user-agent header", func(t *testing.T) {
+		t.Parallel()
+		req := &http.Request{Header: map[string][]string{}}
+		md := userAgentHeadersAnnotator(context.Background(), req)
+		require.NotNil(t, md)
+		assert.Empty(t, md)
+	})
+
+	t.Run("Invalid user-agent formats", func(t *testing.T) {
```

Comment:
Nit: did you mean for this test to run in parallel too?
```suggestion
	t.Run("Invalid user-agent formats", func(t *testing.T) {
		t.Parallel()
```